stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+from odoo.osv import expression
+from odoo.tools.float_utils import float_compare
+
+
+class StockReserveRule(models.Model):
+    """Rules for stock reservations
+
+    A rule is separated in:
+
+    * Check if a rule is applicable, based on the stock move
+      (_is_rule_applicable)
+    * Filter quants that a rule can reserve (_filter_quants)
+    * An advanced removal strategy for the preselected quants (_apply_strategy)
+
+    The first 2 can be cumulative, the removal strategy is exclusive.
+
+    """
+
+    _name = "stock.reserve.rule"
+    _description = "Stock Reservation Rule"
+    _order = "sequence"
+
+    name = fields.Char(string="Description")
+    display_name = fields.Char(compute="_compute_display_name", store=True)
+    sequence = fields.Integer(default=lambda s: s._default_sequence())
+    location_id = fields.Many2one(comodel_name="stock.location", required=True)
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self.env.user.company_id.id,
+    )
+    active = fields.Boolean(default=True)
+
+    # rule applicability
+    rule_domain = fields.Char(
+        string="Rule Domain",
+        default=[],
+        help="Domain based on Stock Moves, to define if the "
+        "rule is applicable or not.",
+    )
+
+    # quants exclusion
+    quant_domain = fields.Char(
+        string="Quants Domain",
+        default=[],
+        help="Filter Quants allowed to be reserved for this location "
+        "and sub-locations.",
+    )
+
+    # advanced removal strategy
+    removal_strategy = fields.Selection(
+        string="Advanced Removal Strategy",
+        selection=[
+            ("default", "Default Removal Strategy"),
+            ("empty_bin", "Empty Bins"),
+            ("packaging", "Prefer Full Packaging"),
```

Comment:
Rename to "Only Full Packaging" (everywhere)

Reply:
reminder